tests/python/BuiltinConfigRegistryTest.py

```diff
@@ -0,0 +1,152 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright Contributors to the OpenColorIO Project.
+
+try:
+    from collections.abc import Iterable
+except ImportError:
+    # Python 2
+    from collections import Iterable
+
+from re import S
+import unittest
+from UnitTestUtils import STRING_TYPES
+
+import PyOpenColorIO as OCIO
+
+class BuiltinConfigRegistryTest(unittest.TestCase):
+    # BuiltinRegistry singleton.
+    REGISTRY = None
+
+    @classmethod
+    def setUpClass(cls):
+        cls.REGISTRY = OCIO.BuiltinConfigRegistry()
+
+    def test_builtin_config_iterable(self):
+        all_names = []
+        for name in self.REGISTRY:
+            self.assertIsInstance(name, STRING_TYPES)
+            self.assertIsInstance(self.REGISTRY[name], STRING_TYPES)
+            all_names.append(name)
+        
+        # All names were iterated over, and __len__ and list() behave.
+        self.assertEqual(len(all_names), len(self.REGISTRY))
+        self.assertListEqual(all_names, list(self.REGISTRY))
+
+        # Test iterator instance.
+        iterator = iter(self.REGISTRY)
+        self.assertIsInstance(iterator, Iterable)
+
+        # Iterator size is available.
+        self.assertEqual(len(iterator), len(self.REGISTRY))
+        
+        # Iterator supports range-based loop and indexing.
+        values = list(iterator)
+        for i in range(len(iterator)):
+            # Item at index matches list result
+            self.assertEqual(iterator[i], values[i])
+            self.assertIsInstance(iterator[i], STRING_TYPES)
+
+    def test_contains(self):
+        # Valid __contains__ for all names.
+        for name in self.REGISTRY:
+            self.assertTrue(name in self.REGISTRY)
+            self.assertFalse(name not in self.REGISTRY)
+
+        # Invalid __contains__ for non-name.
+        with self.assertRaises(OCIO.Exception) as cm:
```

Comment:
I think in this case we would like the `__contains__` magic method not to throw if the name is not found, so the implementation should use the for loop instead of `getBuiltinConfigByName()`.

Reply:
Make sense, I'll change it.